jolt-core/src/poly/commitment/dory.rs

```diff
@@ -0,0 +1,568 @@
+// For now this is needed since we want arkworks Fr
+fn extract_field_coefficients<F: JoltField>(poly: &MultilinearPolynomial<F>) -> Vec<F> {
+    match poly {
+        MultilinearPolynomial::LargeScalars(dense_poly) => dense_poly.Z.clone(),
+        MultilinearPolynomial::U8Scalars(compact_poly) => {
+            compact_poly.coeffs.iter().map(|&c| F::from_u8(c)).collect()
+        }
+        MultilinearPolynomial::U16Scalars(compact_poly) => compact_poly
+            .coeffs
+            .iter()
+            .map(|&c| F::from_u16(c))
+            .collect(),
+        MultilinearPolynomial::U32Scalars(compact_poly) => compact_poly
+            .coeffs
+            .iter()
+            .map(|&c| F::from_u32(c))
+            .collect(),
+        MultilinearPolynomial::U64Scalars(compact_poly) => compact_poly
+            .coeffs
+            .iter()
+            .map(|&c| F::from_u64(c))
+            .collect(),
+        MultilinearPolynomial::I64Scalars(compact_poly) => compact_poly
+            .coeffs
+            .iter()
+            .map(|&c| F::from_i64(c))
+            .collect(),
+    }
+}
+
+/// Dory commitment scheme implementation
+#[derive(Clone, Debug)]
+pub struct DoryCommitmentScheme<F: JoltField, ProofTranscript: Transcript> {
+    _phantom_field: PhantomData<F>,
+    _phantom_transcript: PhantomData<ProofTranscript>,
+}
+
+/// Setup structure containing both prover and verifier parameters.
+/// Dory API exposes an SRS to disk option, but is not (yet) utilized here.
+#[derive(Clone, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DorySetup {
+    pub prover_setup: ProverSetup<ArkBn254Pairing>,
+    pub verifier_setup: VerifierSetup<ArkBn254Pairing>,
+    pub max_log_n: usize,
+}
+
+/// Commitment structure
+#[derive(Clone, Debug, Default, PartialEq, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DoryCommitment {
+    pub commitment: Fq12, // this is GT
+}
+
+/// Proof structure storing the serializable DoryProof data and other data
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DoryProof<F: JoltField> {
+    pub evaluation: Fr,
+    pub opening_point: Vec<Fr>,
+    pub sigma: usize,
+    // Store the serializable DoryProof from the DoryProofBuilder
+    pub dory_proof_data: DoryProofData<ark_bn254::G1Affine, dory::curve::G2AffineWrapper, Fq12>,
+    _phantom: PhantomData<F>,
+}
+
+/// Batched proof structure
+/// @TODO: this is not yet fully implemented.
+#[derive(Clone, Debug, CanonicalSerialize, CanonicalDeserialize)]
+pub struct DoryBatchedProof<F: JoltField> {
+    proofs: Vec<DoryProof<F>>,
+}
+
+impl<F, ProofTranscript> CommitmentScheme<ProofTranscript>
+    for DoryCommitmentScheme<F, ProofTranscript>
+where
+    F: JoltField,
+    ProofTranscript: Transcript,
+{
+    type Field = F;
+    type Setup = DorySetup;
+    type Commitment = DoryCommitment;
+    type Proof = DoryProof<F>;
+    type BatchedProof = DoryBatchedProof<F>;
+
+    fn setup(max_log_n: usize) -> Self::Setup {
```

Comment:
The `CommitmentScheme` trait (and the existing implementations thereof) has this parameter as `max_len`, which is `1 << max_log_n`. That said, I'm down to change it everywhere to this (but maybe rename `max_log_n` to `max_num_vars`)